# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/html/HTMLSelectAccessible.cpp
# Commit: 2236aa837ff6
# Full Hash: 2236aa837ff63d280df545abb6cca31139c6a4b7
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2021-11-16 09:34:25
# Description:
#   Bug 1741014 - Null check container widget in option state change handler. r=morgan
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D131146
# ==============================================================================

diff -r 42e02578ba7c -r 2236aa837ff6 accessible/html/HTMLSelectAccessible.cpp
--- a/accessible/html/HTMLSelectAccessible.cpp	Mon Nov 15 19:23:31 2021 +0000
+++ b/accessible/html/HTMLSelectAccessible.cpp	Mon Nov 15 19:27:28 2021 +0000
@@ -241,7 +241,8 @@
 
   AccStateChangeEvent* event = downcast_accEvent(aEvent);
   if (event && (event->GetState() == states::SELECTED)) {
-    if (!ContainerWidget()->AreItemsOperable()) {
+    LocalAccessible* widget = ContainerWidget();
+    if (widget && !widget->AreItemsOperable()) {
       // Collapsed options' ACTIVE state reflects their SELECT state.
       nsEventShell::FireEvent(this, states::ACTIVE, event->IsStateEnabled(),
                               true);
